src/compiler/docs/generate-json-doc.ts

```diff
@@ -74,19 +74,10 @@ function generateJsDocMembers(cmpMeta: d.ComponentMeta, jsonCmp: d.JsonDocsCompo
         name: memberName
       };
 
-      if (memberMeta.propType === PROP_TYPE.Boolean) {
-        propData.type = 'boolean';
-
-      } else if (memberMeta.propType === PROP_TYPE.Number) {
-        propData.type = 'number';
-
-      } else if (memberMeta.propType === PROP_TYPE.String) {
-        propData.type = 'string';
-
-      } else if (memberMeta.propType === PROP_TYPE.Any) {
-        propData.type = 'any';
-      } else {
-        propData.type = memberMeta.attribType.text;
+      propData = getPropType(memberMeta)
+      
+      if (memberMeta.attribType.optional) {
+        propData.optional = true
```

Comment:
I am not sure I understand this line, is it adding the optional prop to a string?

Reply:
It's appenidng an optional key to the propdata object, which makes that available in the json object produced.  

Happy to pull that out if it's not the way you see this working, but I envisioned using something similar to denote a non required prop in docs I'm generating via the json ouput

Reply:
Oh...wait. Nevermind.  I need to update this later.  That should be propData.type =